tutorials/pretrain_redpajama.md

```diff
@@ -0,0 +1,173 @@
+# Pretrain Llama 2 on RedPajama
+
+This tutorial will walk you through setting up the RedPajama dataset and launching the pretraining script.
+
+## What's RedPajama
+
+[RedPajama](https://github.com/togethercomputer/RedPajama-Data) is an open-source reproduction of the original LLaMA training dataset.
+
+It contains a total of 1.2 trillion tokens, divided into
+
+```text
+Commoncrawl   878B
+C4            175B
+GitHub         59B
+Books          26B
+ArXiv          28B
+Wikipedia      24B
+StackExchange  20B
+```
+
+The [RedPajama repo](https://github.com/togethercomputer/RedPajama-Data) contains the source code for collecting and preparing the dataset, which is Apache 2.0 licensed.
+
+The data itself is licensed according to the original licenses with which its individual parts were released.
+The GitHub datasets are limited to MIT, BSD, or Apache 2.0 repositories.
+
+Along with the full [RedPajama-1T dataset](https://huggingface.co/datasets/togethercomputer/RedPajama-Data-1T),
+the smaller [RedPajama-1T-Sample](https://huggingface.co/datasets/togethercomputer/RedPajama-Data-1T-Sample) 1B sample dataset is also available for development.
+
+You can download the data using git lfs:
+
+```bash
+# The full 1 trillion token dataset:
+# Make sure you have git-lfs installed (https://git-lfs.com): git lfs install
+git clone https://huggingface.co/datasets/togethercomputer/RedPajama-Data-1T data/RedPajama-Data-1T
+```
+
+```bash
+# The 1 billion token subset
+# Make sure you have git-lfs installed (https://git-lfs.com): git lfs install
+git clone https://huggingface.co/datasets/togethercomputer/RedPajama-Data-1T-Sample data/RedPajama-Data-1T-Sample
+```
+
+## Prepare RedPajama for training
+
+The full dataset consists of 2084 `jsonl` files (the sample dataset contains 11). In order to start pretraining lit-gpt
+on it, you need to read, tokenize, and write the data in binary chunks. This will leverage the `PackedDataset`
+streaming dataset that comes with lit-gpt.
+
+Do to so, run
+
+```bash
+python scripts/prepare_redpajama.py --source_path data/RedPajama-Data-1T --checkpoint_dir checkpoints/meta-llama/Llama-2-7b-hf/ --destination_path data/lit-redpajama
+```
+
+or
+
+```bash
+python scripts/prepare_redpajama.py --source_path data/RedPajama-Data-1T-Sample --checkpoint_dir checkpoints/meta-llama/Llama-2-7b-hf/ --destination_path data/lit-redpajama-sample --sample True
+```
+
+for the sample dataset.
+
+In the above we are assuming that you will be using the same tokenizer as used in LLaMA, but any trained [SentencePiece](https://github.com/google/sentencepiece) tokenizer with a 32000 vocabulary size will do here.
+
+The script will take a while to run, so time for :tea:. (The 1B sample script takes about 45 min for the data preparation.)
```

Comment:
Just a question.
Don't know what is the proper way to do this: either put a dot within brackets, or outside, or do not use it at all.
In the `tutorials/neurips_challenge_quickstart.md` you also used a dot within brackets, but I don't know, for me it seems odd. 
Am I wrong? Is it a proper way to do it (like here)?

Reply:
Thanks for the note but this is correct since it's a standalone sentence. The rule is as follows: 
<img width="676" alt="Screenshot 2023-08-24 at 2 55 59 PM" src="https://github.com/Lightning-AI/lit-gpt/assets/5618407/782a99a1-5f4b-4ba4-a895-8f1aac73c376">

Reply:
Thanks a lot! Now I know.
Who would have thought that I can learn English by contributing to OSS 😄.